src/components/patterns/headingList.tsx

```diff
@@ -161,6 +163,11 @@ const HeadingList = () => {
   useEventListener("scroll", debouncedHandleScroll)
   useEventListener("hashchange", handleAnchor)
 
+  // Don't render anything until headings are loaded
+  if (!isLoaded) {
```

Comment:
I wouldn't use a state here because there's still a chance there are no headings on the page. It'll still result in the same errors. just check `headings.length === o`